packages/angular_devkit/build_angular/src/webpack/plugins/javascript-optimizer-plugin.ts

```diff
@@ -84,19 +85,37 @@ export class JavaScriptOptimizerPlugin {
           stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_SIZE,
         },
         async (compilationAssets) => {
+          const cache = compilation.getCache('JavaScriptOptimizerPlugin');
+          const cachedItems: Record<string, ReturnType<typeof cache.getItemCache>> = {};
           const scriptsToOptimize = [];
 
           // Analyze the compilation assets for scripts that require optimization
           for (const assetName of Object.keys(compilationAssets)) {
             if (assetName.endsWith('.js')) {
               const scriptAsset = compilation.getAsset(assetName);
               if (scriptAsset && !scriptAsset.info.minimized) {
-                const { source, map } = scriptAsset.source.sourceAndMap();
+                const { source: scriptAssetSource, name } = scriptAsset;
+                const eTag = cache.getLazyHashedEtag(scriptAssetSource);
+                const cacheItem = cache.getItemCache(name, eTag);
+                const cachedOutput = await cacheItem.getPromise<
+                  { source: sources.Source } | undefined
+                >();
+
+                if (cachedOutput) {
+                  compilation.updateAsset(name, cachedOutput.source, {
+                    minimized: true,
+                  });
+                  continue;
+                }
+
+                const { source, map } = scriptAssetSource.sourceAndMap();
                 scriptsToOptimize.push({
                   name: scriptAsset.name,
                   code: typeof source === 'string' ? source : source.toString(),
                   map,
                 });
+
+                cachedItems[name] = cacheItem;
```

Comment:
I think `cacheItem` could be stored in the object literal pushed into `scriptsToOptimize` above this line.  This should allow removal of the `cachedItems` object map.